### PR TITLE
fix: Improving the infinityLoader implemented

### PR DESCRIPTION
**Pull request by kushthedude**
Refers #3650

### CODE REVIEW

**Comment by iamareebjamal** [comment]
https://www.dictionary.com/browse/improvise

File: app/templates/public/speakers.hbs [hunk 1]
```diff
@@ -4,11 +4,6 @@
       {{public/speaker-item speaker=speaker}}
     </div>
   {{/each}}
-  {{#infinity-loader infinityModel=model}}
-    <div class="center aligned five wide column">
-      <div class="ui loading very padded basic segment">
-      </div>
-    </div>
-    {{infintyModel.reachedInfinity}}
+  {{#infinity-loader infinityModel=model triggerOffset=40 eventDebounce=30}}
```
Comment (iamareebjamal) [inline app/templates/public/speakers.hbs]: Why did you remove the loader?

Comment (kushthedude) [reply app/templates/public/speakers.hbs]: I just removed the spinner, infinityLoader is still there. Removing the spinner reduced the waiting on api host.

Comment (iamareebjamal) [reply app/templates/public/speakers.hbs]: That doesn't make any sense. Please add it back

File: app/templates/public/speakers.hbs [hunk 2]
```diff
@@ -4,11 +4,11 @@
...
-  {{#infinity-loader infinityModel=model}}
-    <div class="center aligned five wide column">
-      <div class="ui loading very padded basic segment">
-      </div>
-    </div>
+  {{#infinity-loader infinityModel=model triggerOffset=40 eventDebounce=30}}
+    <div class="center aligned five wide column">		
+      <div class="ui loading very padded basic segment">		
+      </div>		
+    </div>
```
Comment (iamareebjamal) [inline app/templates/public/speakers.hbs]: Trailing spaces

Comment (kushthedude) [reply app/templates/public/speakers.hbs]: Done

**Comment by iamareebjamal** [comment]
Still the same problem. Pushes the list up instead of down. Tested in Firefox and Chrome. Happens if you are absolutely at the end of the list. Also, have to scroll up and down multiple times before it starts to load stuff. Merging for now, but it is very annoying experience

File: app/templates/public/speakers.hbs [hunk 3]
```diff
@@ -4,7 +4,7 @@
       {{public/speaker-item speaker=speaker}}
     </div>
   {{/each}}
-  {{#infinity-loader infinityModel=model}}
+  {{#infinity-loader infinityModel=model triggerOffset=40 eventDebounce=30}}
```
Comment (iamareebjamal) [inline app/templates/public/speakers.hbs]: Increase it more, still only loads speakers when I am absolutely at the bottom

**Comment by iamareebjamal** [comment]
Also, load at least 12 or 15 speakers

File: app/templates/public/speakers.hbs (hunk 3 above)
Comment (kushthedude) [reply app/templates/public/speakers.hbs]: Done

**Comment by iamareebjamal** [comment]
Well the sticking problem is still there and it only loads items when I am at absolute bottom, so threshold is not working as well. Will see tomorrow what can be done

**Comment by kushthedude** [comment]
> Well the sticking problem is still there and it only loads items when I am at absolute bottom, so threshold is not working as well. Will see tomorrow what can be done

It is a wait of <1 sec TBH. Also, I think micro-location should not be fetched, As we are already displaying micro-location in the session card. What do you suggest?
@iamareebjamal

**Comment by kushthedude** [comment]
> Well the sticking problem is still there and it only loads items when I am at absolute bottom, so threshold is not working as well. Will see tomorrow what can be done

It is a wait of <1 sec TBH. Also, I think micro-location should not be fetched, As we are already displaying micro-location in the session card. What do you suggest?
@iamareebjamal

**Comment by iamareebjamal** [comment]
Let's take microlocation change in next release. About this PR, it doesn't change the experience for me unfortunately

**Comment by kushthedude** [comment]
> Let's take microlocation change in next release. About this PR, it doesn't change the experience for me unfortunately

I tested it just now, It is atleast better than the previous option without offset which was loading 6 speaker at a time 😕 .

File: app/templates/public/speakers.hbs [hunk 4]
```diff
@@ -4,7 +4,7 @@
       {{public/speaker-item speaker=speaker}}
     </div>
   {{/each}}
-  {{#infinity-loader infinityModel=model}}
+  {{#infinity-loader infinityModel=model triggerOffset=400 eventDebounce=50}}
```
Comment (iamareebjamal) [inline app/templates/public/speakers.hbs]: Let's remove the changes here then

Comment (kushthedude) [reply app/templates/public/speakers.hbs]: Sure!
Have we removed the codacy PR check?

Comment (iamareebjamal) [reply app/templates/public/speakers.hbs]: Yes

Comment (iamareebjamal) [inline app/templates/public/speakers.hbs]:
```suggestion
  {{#infinity-loader infinityModel=model}}
```

**Comment by iamareebjamal** [comment]
@kushthedude Release is being made. Do you want this included?

**Comment by kushthedude** [comment]
I suggest including it but I am afk right now, can you please push the required changes, if any?
